frontend/public/components/volumes-table.tsx

```diff
@@ -52,83 +44,107 @@ const ContainerLink: React.FC<ContainerLinkProps> = ({name, pod}) => <span class
+  classNames('col-lg-2', 'col-md-2', 'hidden-sm', 'hidden-xs'),
+  classNames('col-lg-2', 'col-md-2', 'hidden-sm', 'hidden-xs'),
+  classNames('col-lg-2', 'hidden-md', 'hidden-sm', 'hidden-xs'),
+  Kebab.columnClass,
+];
+
+const VolumesTableHeader = () => {
+  return [
+    {
+      title: 'Name', sortField: 'name', transforms: [sortable],
+      props: { className: volumeRoColumnClasses[0]},
+    },
+    {
+      title: 'Mount Path', sortField: 'mountPath', transforms: [sortable],
+      props: { className: volumeRoColumnClasses[1]},
+    },
+    {
+      title: 'SubPath', sortField: 'subPath', transforms: [sortable],
+      props: { className: volumeRoColumnClasses[2]},
+    },
+    {
+      title: 'Type',
+      props: { className: volumeRoColumnClasses[3]},
+    },
+    {
+      title: 'Permissions', sortField: 'readOnly', transforms: [sortable],
+      props: { className: volumeRoColumnClasses[4]},
+    },
+    {
+      title: 'Utilized By', sortField: 'container', transforms: [sortable],
+      props: { className: volumeRoColumnClasses[5]},
+    },
+    {
+      title: '',
+      props: { className: volumeRoColumnClasses[6]},
+    },
+  ];
 };
-VolumeRow.displayName = 'VolumeRow';
+VolumesTableHeader.displayName = 'VolumesTableHeader';
 
-export const VolumesTable: React.FC<VolumesTableProps> = ({podTemplate, heading}) => (
-  <React.Fragment>
-    {heading && <SectionHeading text={heading} />}
-    {_.isEmpty(podTemplate.spec.volumes)
+const VolumesTableRow = ({obj: volume, index, key, style}) => {
+  const type = _.get(getVolumeType(volume.volumeDetail), 'id', '');
+  const loc = getVolumeLocation(volume.volumeDetail);
+  const name = volume.name;
+  const permission = volume.readOnly ? 'Read-only' : 'Read/Write';
+  const { resource } = volume;
+  const pod = getResourceTemplate(resource);
+
+  return (
+    <TableRow id={index} index={index} trKey={key} style={style}>
+      <TableData className={volumeRoColumnClasses[0]}>{name}</TableData>
+      <TableData className={classNames(volumeRoColumnClasses[1], 'co-break-word')}>{volume.mountPath}</TableData>
+      <TableData className={volumeRoColumnClasses[2]}>{volume.subPath}</TableData>
+      <TableData className={volumeRoColumnClasses[3]}>
+        <VolumeIcon kind={type} />
+        <span className="co-break-word">{loc && ` (${loc})`}</span>
+      </TableData>
+      <TableData className={volumeRoColumnClasses[4]}>{permission}</TableData>
+      <TableData className={volumeRoColumnClasses[5]}>
+        {_.get(pod, 'kind') === 'Pod' ? <ContainerLink name={volume.container} pod={pod as PodKind} /> : <div>{volume.container}</div>}
+      </TableData>
+      <TableData className={volumeRoColumnClasses[6]}>
+        <VolumeKebab actions={menuActions} kind={resource.kind} resource={resource} rowVolumeData={volume} />
+      </TableData>
+    </TableRow>
+  );
+};
+VolumesTableRow.displayName = 'VolumesTableRow';
+
+export const VolumesTable = props => {
+  const { resource } = props;
```

Comment:
Does this work?

```suggestion
  const { resource, ...tableProps } = props;
```

and then

```tsx
<Table {...tableProps} ...>
```

Reply:
Yes, it works and the resource is already in the data